Sources/allonet2/PlaceServer.swift

```diff
@@ -318,16 +326,56 @@ public class PlaceServer : AlloSessionDelegate
     {
         switch inter.body
         {
-        case .announce(let version, let avatarDescription):
+
+        case .registerAsAuthenticationProvider:
+            // Reasons this is bad:
+            // - First wins
+            // - Only one provider per place server
+            // - No verification that the client is actually allowed to authenticate others
+            // - A client could authenticate itself
+            if authenticationProvider == nil {
+                authenticationProvider = client
+                client.session.send(interaction: inter.makeResponse(with: .success))
+            } else {
+                throw AlloverseError(domain: PlaceErrorCode.domain, code: PlaceErrorCode.invalidRequest.rawValue,
+                                     description: "Place server already has an authentication provider")
+            }
+
+        case .announce(let version, let authContext, let avatarDescription):
+            // TODO: Since we added authentication, should the version go up?
             guard version == "2.0" else {
                 print("Client \(client.cid) has incompatible version, disconnecting.")
                 client.session.disconnect()
                 return
             }
+
+            if let authenticationProvider, let authenticationId = authenticationProvider.avatar?.id {
+
+                let request = Interaction(type: .request, senderEntityId: Interaction.PlaceEntity,
+                                          receiverEntityId: authenticationId,
+                                          body: .authenticationRequest(authentication: authContext))
+
+                let answer = await authenticationProvider.session.request(interaction: request)
+
+                switch answer.body {
+                case .success: break
+                case .error(let domain, let code, let description): fallthrough
+                default:
+                    // Should we forward the error details back to the client?
+                    let error: InteractionBody = .error(domain: PlaceErrorCode.domain,
+                                                        code: PlaceErrorCode.unauthorized.rawValue,
+                                                        description: "Authentication failed")
+                    client.session.send(interaction: inter.makeResponse(with: error))
+                    client.session.disconnect()
```

Comment:
I think webrtc has a "disconnection reason"; maybe we can inject the error there so we can be more sure it's delivered? maybe write that as a // TODO?